### PR TITLE
Update NodeJS installation instructions using nvm

### PR DESCRIPTION
**Pull request by raupie**
NODE_VERSION env var triggers automatic nodejs installation which triggers a bash error

### CODE REVIEW

**Comment by raupie** [comment]
When we updated this snippet to use environment variables to store the version numbers, it allowed the install.sh script to auto install NodeJS. This is cool, but it also triggers a bash error. Plus `nvm install $NODE_VERSION` was repeating the external curl to download the nodejs package resulting in a longer build time.

```
=> Installing Node.js version v14.8.0
        Downloading and installing node v14.8.0...
        W: Downloading https://nodejs.org/dist/v14.8.0/node-v14.8.0-linux-x64.tar.xz...
######################################################################## 100.0%
        W: Computing checksum with sha256sum
        W: Checksums matched!
        Now using node v14.8.0 (npm v6.14.7)
        Creating default alias: default -> v14.8.0 *
        W: dash: 163: [: v14.8.0: unexpected operator
        W: Failed to install Node.js v14.8.0
        => Close and reopen your terminal to start using nvm or run the following to use it now:
```

```
  CURRENT_NVM_NODE="$(nvm_version current)"
  if [ "$(nvm_version "$NODE_VERSION_LOCAL")" == "$CURRENT_NVM_NODE" ]; then
```

Since NodeJS was already installed, `nvm_version current` is outputting `system` thus failing this conditional statement.

https://github.com/nvm-sh/nvm/blob/e01060fa2c331d77e306a90288b759010e14758e/install.sh#L147

I propose we rename the NodeJS version env var to `NODEJS_VERSION` to bypass the automatic installation then let `nvm install $NODEJS_VERSION` handle the installation part.

**Comment by raupie** [comment]
https://github.com/nvm-sh/nvm/ also looks to be the new Git Repo for the nvm install script.

**Comment by raupie** [comment]
https://github.com/nvm-sh/nvm/issues/2319

I believe there is an issue with the string comparison for == in the nvm git repo. I submitted an issue for it. Doesn't seem like `==` is a POSIX safe conditional.

**Comment by raupie** [comment]
nvm maintainers suggest running install.sh with bash instead of dash. 

Tested and it works.

**Comment by raupie** [comment]
squashed + rebased + pushed

File: docs/src/languages/nodejs/nvm.md [hunk 1]
```diff
@@ -17,15 +17,14 @@ variables:
     env:
         # Update these for your desired NVM and Node versions.
         NVM_VERSION: v0.35.2
```
Comment (RudyWeber) [inline docs/src/languages/nodejs/nvm.md:19]: Maybe we can also update the `NVM_VERSION` at the same time. The latest one is `v0.36.0` now.

Comment (raupie) [reply docs/src/languages/nodejs/nvm.md]: Yes, why not!

**Review by RudyWeber — CHANGES_REQUESTED** [review]

File: docs/src/languages/nodejs/nvm.md [hunk 2]
```diff
@@ -16,16 +16,15 @@ Installing NVM is done in the build hook of your `.platform.app.yaml`, which som
...
+        NODE_VERSION: v14.13.1
 
 hooks:
     build: |
         unset NPM_CONFIG_PREFIX
-        curl -o- https://raw.githubusercontent.com/creationix/nvm/$NVM_VERSION/install.sh | dash
         export NVM_DIR="$PLATFORM_APP_DIR/.nvm"
+        curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/$NVM_VERSION/install.sh | bash
         [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
-        nvm install $NODE_VERSION
```
[the remark this answers is not in the document]
Comment (raupie) [reply docs/src/languages/nodejs/nvm.md:28]: Right, that's what broke originally, when the `$NODE_VERSION` is set, then it triggers that auto install function.

[the remark this answers is not in the document]
Comment (raupie) [reply docs/src/languages/nodejs/nvm.md:28]: the order of the `export NVM_DIR="$PLATFORM_APP_DIR/.nvm"` is important before the curl.

[the remark this answers is not in the document]
Comment (raupie) [reply docs/src/languages/nodejs/nvm.md:28]: `# install.sh will automatically install NodeJS based on the presence of $NODE_VERSION`

thoughts?

[the remark this answers is not in the document]
Comment (RudyWeber) [reply docs/src/languages/nodejs/nvm.md:28]: Sounds good to me!

[the remark this answers is not in the document]
Comment (raupie) [reply docs/src/languages/nodejs/nvm.md:28]: update coming

**Review by RudyWeber — APPROVED** [review]

**Pull request merged**